app/controllers/popup_controller.rb

```diff
@@ -119,4 +146,41 @@ def reviewer_details_popup
     @user = User.find(@userid)
     @id = params[:assignment_id]
   end
+
+  def calculate_percentages(assign_id)
+    #single_response = Response.where(id: record_id)
+    #mapped_response = ResponseMap.where(id: single_response[0].map_id)
+    review_maps = ResponseMap.where(reviewed_object_id: assign_id)
+    keys = [[0.00, 0.00, 0.00, 0.00], [0.00, 0.00, 0.00, 0.00], [0.00, 0.00, 0.00, 0.00], [0.00, 0.00, 0.00, 0.00]]
+    response_count = [0.00, 0.00, 0.00, 0.00, 0.00, 0.00]
+    word_counter = [0, 0, 0, 0, 0, 0]
+    suggestive_count = [0, 0, 0, 0, 0, 0]
+    problem_count = [0, 0, 0, 0, 0, 0]
+    offensive_count = [0, 0, 0, 0, 0, 0]
```

Comment:
It seems to me that if we want to add another other metrics, say, unique words, we have to change the code here. Is there any metaprogramming can be possibly used here?